sapp/pipeline/add_features.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import logging
+from typing import List, Optional, Set, Tuple
+
+from . import DictEntries, PipelineStep, Summary
+
+log: logging.Logger = logging.getLogger("sapp")
+
+
+class AddFeatures(PipelineStep[DictEntries, DictEntries]):
+    """Optional pipeline that attaches given features to all issues.
+    If the features list is empty, the input is simply passed without
+    change as the output"""
+
+    def __init__(self, features: Optional[List[str]]) -> None:
+        super().__init__()
+        # pyre-fixme[6]: Expected `typing.Iterable[Variable[_T]]` for `set.__init__`
+        # but got `Optional[List[str]]`
+        self.features: Set[str] = set(features)
```

Comment:
We definitely shouldn't be suppressing an error here. Either `features` can't be `None`, and we should drop the `Optional` annotation, or it can be and this code will throw an exception when you do `set(None)`. Handling the optional like this
```suggestion
        self.features: Set[str] = set(features || [])
```

Reply:
Used or and it seems working :)